### PR TITLE
Propagate bearer token when fetching changeset from architect

### PR DESCRIPTION
**Pull request by tgeens**

### CODE REVIEW

File: scribe-configurations/scribe-app/src/main/java/eu/xenit/contentcloud/scribe/app/ScribeApplication.java [hunk 1]
```diff
@@ -45,6 +55,32 @@ RestTemplateCustomizer hypermediaRestTemplateCustomizer(HypermediaRestTemplateCo
...
+     *
+     * @return the {@link RestTemplateCustomizer} that propagates the bearer token from the current request
+     *
+     * @see <a href="https://docs.spring.io/spring-security/reference/servlet/oauth2/resource-server/bearer-tokens.html#_resttemplate_support">Spring Security - OAuth 2.0 Bearer Tokens</a>
+     */
+    @Bean
+    RestTemplateCustomizer bearerTokenPropagationRestTemplateCustomizer() {
+        return restTemplate -> restTemplate.getInterceptors().add((request, body, execution) -> {
+
+                Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
```
Comment (vierbergenlars) [inline scribe-configurations/scribe-app/src/main/java/eu/xenit/contentcloud/scribe/app/ScribeApplication.java:70]: I don't think we should indiscriminately forward our bearer token with any `RestTemplate` request. We should check in some way which host we are targetting.

Comment (tgeens) [reply scribe-configurations/scribe-app/src/main/java/eu/xenit/contentcloud/scribe/app/ScribeApplication.java:70]: We actually do. The property `scribe.allow-list` controls that.

But your argument stands: what if we later on use `RestTemplate` somewhere else ? We risk leaking our access token to a third party at that point.

Comment (vierbergenlars) [reply scribe-configurations/scribe-app/src/main/java/eu/xenit/contentcloud/scribe/app/ScribeApplication.java:70]: Yes, I was thinking exactly about that case: when using RestTemplate somewhere else.

I think we should maybe key the adding of authentication credentials to that allow list, or add the credentials at the actual call-site.

Comment (rschev) [reply scribe-configurations/scribe-app/src/main/java/eu/xenit/contentcloud/scribe/app/ScribeApplication.java:70]: I think adding credentials at the call site is the clearer (and more maintainable) option.

**Review by vierbergenlars — APPROVED** [review]

File: scribe-configurations/scribe-app/src/main/java/eu/xenit/contentcloud/scribe/app/ScribeApplication.java (hunk 1 above)
Comment (tgeens) [reply scribe-configurations/scribe-app/src/main/java/eu/xenit/contentcloud/scribe/app/ScribeApplication.java]: I would argue the other way around - moving the `scribe.allow-list` check to an interceptor is more maintainable ;-)

**Pull request merged**